src/generic/GenericTypes.jl

```diff
@@ -798,6 +798,42 @@ end
 
 const PuiseuxSeriesElem{T} = Union{PuiseuxSeriesRingElem{T}, PuiseuxSeriesFieldElem{T}} where T <: RingElement
 
+###############################################################################
+#
+#   AbsMSeriesRing / AbsMSeries
+#
+###############################################################################
+
+mutable struct AbsMSeriesRing{T <: RingElement} <: AbstractAlgebra.MSeriesRing{T}
+   base_ring::Ring
+   poly_ring::MPolyRing{T}
+   prec_max::Vector{Int}
+   sym::Vector{Symbol}
+
+   function AbsMSeriesRing{T}(R::Ring, poly_ring::MPolyRing{T}, prec::Vector{Int}, s::Vector{Symbol}, cached::Bool = true) where T <: RingElement
+      if cached && haskey(AbsMSeriesID, (R, prec, s))
+         return AbsMSeriesID[R, prec, s]::AbsMSeriesRing{T}
+      else
+         z = new{T}(R, poly_ring, prec, s)
+         if cached
+            AbsMSeriesID[R, prec, s] = z
+         end
+         return z
+      end
+   end
+end
+
+const AbsMSeriesID = Dict{Tuple{Ring, Vector{Int}, Vector{Symbol}}, Ring}()
+
+mutable struct AbsMSeries{T <: RingElement} <: AbstractAlgebra.AbsMSeriesElem{T}
+   poly::MPoly{T}
+   prec::Vector{Int}
+   parent::AbsMSeriesRing{T}
+
+   AbsMSeries{T}(p::MPoly{T}, prec::Vector{Int}) where T <: RingElement = new{T}(p, prec)
+   AbsMSeries{T}(a::AbsMSeries{T}) where T <: RingElement = a
```

Comment:
Hm, we don't do this for other ring elements. Do we need this for this type?

Reply:
Could you be more specific. I am not sure what your comment refers to.

Reply:
I don't think we have `typeof(x)(x) === x` for any other type defined

```julia
julia> Zx, x = ZZ["x"]
(Univariate Polynomial Ring in x over Integers, x)

julia> typeof(x)(x)
ERROR: MethodError: no method matching AbstractAlgebra.Generic.Poly{BigInt}(::AbstractAlgebra.Generic.Poly{BigInt})
Closest candidates are:
  AbstractAlgebra.Generic.Poly{T}(::T) where T<:RingElement at /home/bla/.julia/packages/AbstractAlgebra/di40H/src/generic/GenericTypes.jl:357
  AbstractAlgebra.Generic.Poly{T}() where T<:RingElement at /home/bla/.julia/packages/AbstractAlgebra/di40H/src/generic/GenericTypes.jl:349
  AbstractAlgebra.Generic.Poly{T}(::Vector{T}) where T<:RingElement at /home/bla/.julia/packages/AbstractAlgebra/di40H/src/generic/GenericTypes.jl:351
Stacktrace:
 [1] top-level scope
   @ REPL[11]:1
```

Reply:
Or are you just referring to omitting line 834? If so, yes I can remove it. It's just a cut'n'paste from AbsSeries probably.

Reply:
Orobably I clicked the wrong thing. Sorry for the confusion. Yes I mean 834.

Reply:
I've removed it. Wasn't needed.